torchaudio/functional/functional.py

```diff
@@ -1379,7 +1382,10 @@ def _get_sinc_resample_kernel(
         kernels.append(kernel)
 
     scale = base_freq / orig_freq
-    return torch.stack(kernels).view(new_freq, 1, -1).mul_(scale), width
+    kernels = torch.stack(kernels).view(new_freq, 1, -1).mul_(scale)
+    if dtype is None:
+        kernels = kernels.to(dtype=torch.float32)
```

Comment:
It might be better to just return the kernel and do the dtype and device cast after the callsite, since you're not using dtype outside of arange.

Reply:
following offline discussion, it is fine to convert to this generally "default" type prior to returning the kernel to the call from transforms